internal/dataplane/parser/golden_test.go

```diff
@@ -0,0 +1,300 @@
+package parser_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/blang/semver/v4"
+	"github.com/samber/lo"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/yaml"
+
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/deckgen"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane/parser"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/store"
+)
+
+var (
+	// updateGolden tells whether to update golden files using the current output of the parser.
+	updateGolden = os.Getenv("UPDATE_GOLDEN") == "true"
+
+	// defaultKongVersion is the default Kong version to use in tests. Can be overridden in a test case.
+	defaultKongVersion = semver.MustParse("3.3.0")
+
+	// defaultFeatureFlags is the default set of feature flags to use in tests. Can be overridden in a test case.
+	defaultFeatureFlags = func() parser.FeatureFlags {
+		return parser.FeatureFlags{
+			CombinedServiceRoutes: true,
+			RegexPathPrefix:       true,
+		}
+	}
```

Comment:
Can't we use the defaults from https://github.com/Kong/kubernetes-ingress-controller/blob/45948a63c178fab2161b268899c4d307b420c7d1/internal/manager/featuregates/feature_gates.go#L72-L87 ?